### PR TITLE
Add: Tutorial for migrating from setup.py to pyproject.toml using Hatch

### PR DESCRIPTION
**Pull request by miguelalizo**
Addresses #210 by creating the tutorial for migrating from setup.py to pyproject.toml using Hatch.

### CODE REVIEW

**Review by willingc — APPROVED** [review]
Looking good. Thanks @miguelalizo

File: tutorials/setup-py-to-pyproject-toml.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+:class: tip
+
+In this lesson you will learn:
+
+1. The process of using Hatch to transition to using `pyproject.toml` for projects that already have a `setup.py` defined.
+:::
+
+## What is Hatch?
+
+See [Get to know Hatch](get-to-know-hatch.md).
```
Comment (ucodery) [inline tutorials/setup-py-to-pyproject-toml.md:22]: This section isn't very helpful unless we give at least a minimal definition. "hatch is a python build tool"

File: tutorials/setup-py-to-pyproject-toml.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Prerequisites
+
+Before we begin, ensure that you have Hatch installed on your system. You can install it via pip:
+
+```bash
+pip install hatch
```
Comment (ucodery) [inline tutorials/setup-py-to-pyproject-toml.md:29]: The rest of this guide settled on a particular idiom
```bash
python -m pip install hatch
```

File: tutorials/setup-py-to-pyproject-toml.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+2. **Initialize Hatch**: Run the following command to initialize Hatch in your project directory:
+
+   ```bash
+   hatch new --init
+   ```
+
+3. **Review and Customize**: After running the previous command, Hatch will automatically generate a `pyproject.toml` file based on your existing project configuration. Take some time to review the contents of the generated `pyproject.toml` file. You may want to customize certain settings or dependencies based on your project's requirements (see [pyproject.toml tutorial](pyproject-toml.md) for more information about the `pyproject.toml`).
+
+4. **Verify**: Verify that the `pyproject.toml` file accurately reflects your project configuration and dependencies. You can manually edit the file if needed, but be cautious and ensure that the syntax is correct.
```
Comment (ucodery) [inline tutorials/setup-py-to-pyproject-toml.md:90]: ISTM this guide isn't done yet. Migrating to pyproject.toml implies that setup.py will go away. I think there are two more steps:

5. delete setup.py

6. test - build the project with only pyproject.toml, it should not only build but pass any preexisting tests.

File: tutorials/setup-py-to-pyproject-toml.md (hunk 2 above)
Comment (ucodery) [reply tutorials/setup-py-to-pyproject-toml.md:29]: Actually, for CLI tools like this we go with `pipx install hatch`

Comment (miguelalizo) [reply tutorials/setup-py-to-pyproject-toml.md:29]: Perfect, thanks for the clarification. Just updated it to use pipe

File: tutorials/setup-py-to-pyproject-toml.md (hunk 1 above)
Comment (miguelalizo) [reply tutorials/setup-py-to-pyproject-toml.md]: Good call! I just added a short description of what hatch is and added a admonition for calling out the get to know hatch page.

File: tutorials/setup-py-to-pyproject-toml.md (hunk 3 above)
Comment (miguelalizo) [reply tutorials/setup-py-to-pyproject-toml.md:97]: Great idea. I expanded on this and added 3 more steps: 

- Step 5: Delete setup.py

- Step 6: Test Build

- Step 7: Test Existing Functionality

File: tutorials/setup-py-to-pyproject-toml.md [hunk 4]
```diff
@@ -88,3 +95,25 @@ Now, let's walk through the steps to use Hatch to create a `pyproject.toml` file
...
 
 4. **Verify**: Verify that the `pyproject.toml` file accurately reflects your project configuration and dependencies. You can manually edit the file if needed, but be cautious and ensure that the syntax is correct.
+
+### Step 5: Delete setup.py
+
+Since we're migrating to using `pyproject.toml` exclusively, the `setup.py` file becomes unnecessary. You can safely delete it from your project directory.
+
+```bash
+rm setup.py
+```
```
Comment (ucodery) [inline tutorials/setup-py-to-pyproject-toml.md:105]: IMO we don't need to add bash examples for every step. Especially as some readers won't want to do everything in the terminal.

File: tutorials/setup-py-to-pyproject-toml.md [hunk 5]
```diff
@@ -88,3 +95,25 @@ Now, let's walk through the steps to use Hatch to create a `pyproject.toml` file
 3. **Review and Customize**: After running the previous command, Hatch will automatically generate a `pyproject.toml` file based on your existing project configuration. Take some time to review the contents of the generated `pyproject.toml` file. You may want to customize certain settings or dependencies based on your project's requirements (see [pyproject.toml tutorial](pyproject-toml.md) for more information about the `pyproject.toml`).
 
 4. **Verify**: Verify that the `pyproject.toml` file accurately reflects your project configuration and dependencies. You can manually edit the file if needed, but be cautious and ensure that the syntax is correct.
+
+### Step 5: Delete setup.py
```
Comment (ucodery) [inline tutorials/setup-py-to-pyproject-toml.md:99]: This guide switches from an ordered list to numbered headers. Let's stick to one style.

**Review by ucodery — CHANGES_REQUESTED** [review]
Looking really good!

File: tutorials/setup-py-to-pyproject-toml.md (hunk 5 above)
Comment (miguelalizo) [reply tutorials/setup-py-to-pyproject-toml.md]: Good catch! Thanks for pointing it out. Just fixed it to be consistent

File: tutorials/setup-py-to-pyproject-toml.md (hunk 4 above)
Comment (miguelalizo) [reply tutorials/setup-py-to-pyproject-toml.md]: I agree. I removed some of these and left only a couple of hatch specific ones.

**Review by ucodery — APPROVED** [review]
Thanks for the edit

**Review by lwasser — APPROVED** [review]
Two approvals. we can merge this as well! in the future maybe we can have a demo repo with a setup.py file that people can play with. but for now let's get this merged!! many many thanks everyone for your effort on this!!

**Pull request merged**